isis/src/base/objs/Cube/Cube.cpp

```diff
@@ -578,43 +548,33 @@ namespace Isis {
     }
 
     if (m_dataFile) {
-      if (m_storesDnData && !m_dataFile->open(QIODevice::Truncate | QIODevice::ReadWrite)) {
+      if (labelsAttached() != ExternalLabel && !m_dataFile->open(QIODevice::Truncate | QIODevice::ReadWrite)) {
         QString msg = "Failed to create [" + m_dataFile->fileName() + "]. ";
         msg += "Verify the output path exists and you have permission to write to the path.";
         cleanUp(false);
         throw IException(IException::Io, msg, _FILEINFO_);
       }
-      else if (!m_storesDnData && !m_dataFile->open(QIODevice::ReadWrite)) {
+      else if (labelsAttached() == ExternalLabel && !m_dataFile->open(QIODevice::ReadOnly)) {
         QString msg = "Failed to open [" + m_dataFile->fileName() + "] for reading. ";
         msg += "Verify the output path exists and you have permission to read from the path.";
         cleanUp(false);
         throw IException(IException::Io, msg, _FILEINFO_);
       }
     }
 
-    bool dataAlreadyOnDisk = m_storesDnData ? false : true;
+    bool dataAlreadyOnDisk = (labelsAttached() == ExternalLabel);
 
     if (m_format == Bsq) {
       m_ioHandler = new CubeBsqHandler(dataFile(), m_virtualBandList, realDataFileLabel(),
                                        dataAlreadyOnDisk);
     }
-    else if (m_format == Tile) {
+    else {
       m_ioHandler = new CubeTileHandler(dataFile(), m_virtualBandList, realDataFileLabel(),
                                         dataAlreadyOnDisk);
     }
```

Comment:
we are not handling gdal in this PR?

Reply:
Correct, I didn't handle gdal in that PR. Just straightened out the attached vs detached vs external